connectors/src/lib/data_sources.ts

```diff
@@ -728,6 +727,13 @@ export async function upsertTableFromCsv({
 
   const now = new Date();
 
+  if (new Blob([tableCsv]).size > MAX_CSV_SIZE) {
```

Comment:
I've added a size check here before sending the request to front, to avoid creating unnecessary requests with huge bodies